File: leetcode/easy/1281/1281.cpp
/*
 * Runtime: 0 ms, faster than 100.00%
 * Memory Usage: 6.2 MB, less than 64.71%
 */
class Solution {
  public:
    int subtractProductAndSum(int n) {
      int p = 1;
      int s = 0;

      while(n>0) 
      {
        int t = n%10;
        n = n/10;

        p *= t;
        s += t;
      }

      return p - s;
    }
};

File: leetcode/easy/1281/1281.rb
# Runtime: 44 ms, faster than 83.10%
# Memory Usage: 209.8 MB, less than 7.04%
def subtract_product_and_sum(n)
  p = 1
  s = 0

  while n > 0 do
    t = n % 10
    n /= 10

    p *= t
    s += t
  end

  p - s
end
